docs: document NoExecute taint risks and add admission warning (#120)
kubernetes-sigs/node-readiness-controller: docs/book/src/user-guide/getting-started.md

Reviewer: ```suggestion
```yaml
apiVersion: readiness.node.x-k8s.io/v1alpha1
kind: NodeReadinessRule
metadata:
  name: nfs-storage-readiness-rule
spec:
  conditions:
    - type: "csi.example.net/NodePluginRegistered"
    requiredStatus: "True"
    - type: "csi.example.net/BackendReachable"
    requiredStatus: "True"
    - type: "DiskPressure"
    requiredStatus: "False"
  taint:
    key: "readiness.k8s.io/example.net/nfs-unhealthy"
    effect: "NoSchedule"
  enforcementMode: "bootstrap-only"
  nodeSelector:
     matchLabels:
```

Reviewer: ```suggestion
- type: "csi.example.net/NodePluginRegistered"
```

update conditions to take subdomain and use consistent examples.

Reviewer: let us leave Cluster Autoscaler for now

Author: any changes needed in the PR for this?

| `nodeSelector` | Label selector to target specific nodes | No |
| `dryRun` | Preview changes without applying them | No |

### Enforcement Modes

#### Bootstrap-only Mode
- Removes bootstrap taint when conditions are first satisfied
- Marks completion with node annotation
- Stops monitoring after successful removal (fail-safe)

> - Ideal for one-time setup conditions (storage, installing node daemons e.g: security agent or kernel-module update)

Reviewer: ```suggestion
- Ideal for one-time setup conditions (installing node daemons e.g: security agent or kernel-module update)
```

kubectl get nodereadinessrule <rule-name> -o jsonpath='{.status.dryRunResults}'
```

### Rule Validation and Constraints

#### NoExecute Taint Effect Warning

**`NoExecute` with `continuous` enforcement mode will evict existing workloads when conditions fail.**

> If a critical component becomes temporarily unavailable (e.g., CNI daemon restart), all pods without matching tolerations are immediately evicted from the node. Use `NoSchedule` to prevent new scheduling without disrupting running workloads.

Reviewer: ```suggestion
If a readiness condition on the node is failing temporarily (eg., the component restarted), all pods without matching tolerations are immediately evicted from the node, if configured with a `NoExecute` taint. Use `NoSchedule` to prevent new scheduling without disrupting running workloads.
```

> The admission webhook warns when using `NoExecute`. 

Reviewer: Did you get a chance to test your changes w/ warnings? You may need to follow this PR: https://github.com/kubernetes-sigs/node-readiness-controller/pull/122 for testing the webhook.

nit / optional: If you get a chance to verify it, you could also add the warning here as response to kubectl.
